admin/app/helpers/solidus_admin/permission_sets_helper.rb

```diff
@@ -0,0 +1,109 @@
+        when /ConfigurationDisplay/i
+          grouped_permissions[:configuration] ||= []
+          grouped_permissions[:configuration] << [view_string, permission.id]
+        when /ConfigurationManagement/i
+          grouped_permissions[:configuration] ||= []
+          grouped_permissions[:configuration] << [edit_string, permission.id]
+        when /UserDisplay/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [view_string, permission.id]
+        when /UserManagement/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [edit_string, permission.id]
+        when /RestrictedStockDisplay/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [view_string, permission.id]
+        when /RestrictedStockManagement/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [edit_string, permission.id]
+        when /StockDisplay/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [view_string, permission.id]
+        when /StockManagement/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [edit_string, permission.id]
+        when /ProductDisplay/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [view_string, permission.id]
+        when /ProductManagement/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [edit_string, permission.id]
+        else
+          grouped_permissions[:other] ||= []
+          grouped_permissions[:other] << [permission.name, permission.id]
+        end
+      end
+    end
+
+    # Helper for rendering a set of permission checkboxes.
+    #
+    # @param form [ActionView::Helpers::FormBuilder] The form builder object
+    #   used to generate form elements.
+    # @param title [String] The desired component title.
+    # @param permissions [Array<Array<String, String>>] An array of permission
+    #   pairs, where each pair consists of:
+    #   - A display name for the permission (String).
+    #   - An identifier for the permission (String) which is used as the value
+    #     for the checkbox input.
+    #
+    # @return [String, nil] Returns the generated HTML as a string if
+    #   permissions are present, otherwise returns nil.
+    #
+    # @example
+    #   <%= permission_set_component(
+    #     form: f,
+    #     title: 'Orders',
+    #     permissions: permission_set_options[:order]
+    #   ) %>
+    #   <%= permission_set_component(
+    #     form: f,
+    #     title: 'Products',
+    #     permissions: permission_set_options[:product]
+    #   ) %>
+    def permission_set_component(form:, title:, permissions:)
+      return if permissions.blank?
+
+      content_tag :span, class: "p-2 flex hover:bg-gray-50 rounded-md" do
```

Comment:
I know that we have this a lot in the new admin, but I am not a big fan of having all these Tailwind classes flying around all over the admin now. Can we create a view component instead, so it is at least in a place that could be configured some day?

Reply:
Fixed :)